elements/lisk-p2p/src/peer_selection.ts

```diff
@@ -123,5 +123,38 @@ export const selectPeersForConnection = (
 		return [];
 	}
 
-	return shuffle(input.peers).slice(0, input.peerLimit);
+	if (
+		input.peerLimit === undefined ||
+		input.peerLimit >= input.triedPeers.length + input.newPeers.length
+	) {
+		return [...input.newPeers, ...input.triedPeers];
+	}
+
+	if (input.triedPeers.length === 0 && input.newPeers.length === 0) {
+		return [];
+	}
+
+	const x =
+		input.triedPeers.length / (input.triedPeers.length + input.newPeers.length);
+	// tslint:disable-next-line: no-magic-numbers
+	const r = Math.max(x, 0.5);
```

Comment:
Can we make this value, that comes from the LIP, a constant with possibly a meaningful name or if naming it meaningfully is too complex add a link to the lip section describing the selection of this value? Just for the sake of clarity :)